WIP: Serach for an existing key by fingerprint before attempting upload

am97 opened the pull request:
fixes #49

With this fix, the creation of two machines as described in #49 works. The removal is also successful:

```
am@localhost:~/go/src/github.com/am97/docker-machine-driver-hetzner$ docker-machine rm machine1
About to remove machine1
WARNING: This action will delete both local reference and remote instance.
Are you sure? (y/n): y
(machine1)  -> Destroying server machine1[7976813] in...
(machine1)  -> Destroying SSHKey am@am[2167977]...
Successfully removed machine1
am@localhost:~/go/src/github.com/am97/docker-machine-driver-hetzner$ docker-machine rm machine2
About to remove machine2
WARNING: This action will delete both local reference and remote instance.
Are you sure? (y/n): y
(machine2)  -> Destroying server machine2[7976887] in...
(machine2)  -> SSH key does not exist anymore
Successfully removed machine2
```

JonasProgrammer commented on driver.go:
```diff
@@ -241,14 +241,29 @@ func (d *Driver) Create() error {
-		keyopts := hcloud.SSHKeyCreateOpts{
-			Name:      d.GetMachineName(),
-			PublicKey: string(buf),
+		// Try to find a key with the same Fingerprint
```
In general: given the amount of logic here, I'd almost opt to put this in a separate function

JonasProgrammer commented on driver.go:
```diff
@@ -241,14 +241,29 @@ func (d *Driver) Create() error {
+		fp := ssh.FingerprintLegacyMD5(pubkey)
+
+		key, _, err := d.getClient().SSHKey.GetByFingerprint(context.Background(), fp)
 		if err != nil {
```
Did you get key and err mixed up here?

am97 replied:
You are right, the condition should be `if key == nil`

JonasProgrammer commented on driver.go:
```diff
@@ -241,14 +241,29 @@ func (d *Driver) Create() error {
+		key, _, err := d.getClient().SSHKey.GetByFingerprint(context.Background(), fp)
 		if err != nil {
-			return errors.Wrap(err, "could not create ssh key")
+			log.Debugf("SSH key not found in Hetzner. Uploading...")
```
Personally I think that this is actually useful information for normal users, as you should refrain from altering auto-uploaded keys manually instead of using `docker-machine`.
So I'd opt to use `Infof` instead of `Debugf` here.

am97 commented:
Thanks for your feedback. I moved some logic to `getRemoteKeyWithSameFingerprint()` (the name is a bit long, but I don't have other ideas), fixed the condition and replaced `log.Debugf` with `log.Infof` to inform users that SSH key is 
being uploaded.

Feel free to edit the PR if you want :)

JonasProgrammer reviewed (approved):
LGTM
